api/src/main/java/net/md_5/bungee/command/PlayerCommand.java

```diff
@@ -31,20 +28,9 @@ public PlayerCommand(String name, String permission, String... aliases)
     public Iterable<String> onTabComplete(CommandSender sender, String[] args)
     {
         final String lastArg = ( args.length > 0 ) ? args[args.length - 1].toLowerCase( Locale.ROOT ) : "";
-        return Iterables.transform( Iterables.filter( ProxyServer.getInstance().getPlayers(), new Predicate<ProxiedPlayer>()
-        {
-            @Override
-            public boolean apply(ProxiedPlayer player)
-            {
-                return player.getName().toLowerCase( Locale.ROOT ).startsWith( lastArg );
-            }
-        } ), new Function<ProxiedPlayer, String>()
-        {
-            @Override
-            public String apply(ProxiedPlayer player)
-            {
-                return player.getName();
-            }
-        } );
+        return ProxyServer.getInstance().getPlayers().stream()
+                .map( player -> player.getName() )
+                .filter( name -> name.toLowerCase( Locale.ROOT ).startsWith( lastArg ) )
+                .collect( Collectors.toList() );
```

Comment:
This is actually worse than Guava because Iterables.transform and ilk are lazy, this is not.

Reply:
This might help you create an Iterable result without creating an ArrayList instance:
https://stackoverflow.com/questions/59869486/how-can-i-turn-a-stream-into-an-iterable

Reply:
Guava doesn't make an ArrayList. You do. Making an ArrayList is unecessarily slow. The Guava code can of course be lambdafied, but overall it is a better implementation (even than what Black-Hole linked which is hacky).

Reply:
i think i found a nice way